tests/Properties.hs

```diff
@@ -1241,17 +1241,17 @@ prop_initsBB xs = inits xs == map P.unpack (P.inits (P.pack xs))
 prop_tailsBB xs = tails xs == map P.unpack (P.tails (P.pack xs))
 
 prop_findSubstringsBB s x l
-    = C.findSubstrings (C.pack p) (C.pack s) == naive_findSubstrings p s
+    -- coerce to 8-bit alphabet to avoid false-positives
+    = let s8 = C.unpack $ C.pack s
+          _ = l :: Int
+          _ = x :: Int
```

Comment:
Just curious, what does this idiom mean? Is it just to fix types or `l` and `x`? We can probably just throw a type signature for `prop_findSubstringsBB` for this purpose.

Also, I suggest to keep formating of this `prop_findSubstringsBB` consistent with the next one, `prop_findSubstringBB`.

Reply:
Well, I just moved that code without changing it.  Your guess about the syntax looks right to me.

Reply:
Could you do that little cleanup Andrew suggested then I’ll apply this fix to master and then figure out build surgery

Reply:
Done.  I force-pushed an updated and cleaned up set of commits, and perhaps also fixed a less likely, but still possible in principle failure in `findSubstring` that is essentially the same as the one in `findSubstrings`.  When the input is not just 8-bit bytes the `pack` is not injective and there are more matches in the range than in the domain.